### DISCUSSION TITLE
NATS.ws and React Native Checklist

### DISCUSSION THREAD

**Discussion by aricart** [nats-io/nats.ws#138] [General]
To use nats.ws with React Native:

- Use the current version of React Native
- You'll need two polyfills:
  - https://www.npmjs.com/package/react-native-url-polyfill
  - https://www.npmjs.com/package/text-encoding-polyfill
- In some cases you may need to change `pingInterval` to something like `120000` 

Special thanks to @machester4

**Comment by machester4** [comment]
It is also necessary add a **`metro.config.js`** file in the root directory of your React Native project and set `resolver.extraNodeModules` with https://github.com/parshap/node-libs-react-native to make all node modules available like util.

FYI https://gist.github.com/parshap/e3063d9bf6058041b34b26b7166fd6bd
